OrbitGl/ModuleData.h

```diff
@@ -9,35 +9,57 @@
 #include <memory>
 #include <string>
 #include <utility>
+#include <vector>
 
+#include "absl/container/flat_hash_map.h"
 #include "absl/strings/str_format.h"
+#include "absl/synchronization/mutex.h"
+#include "capture_data.pb.h"
 #include "module.pb.h"
+#include "symbol.pb.h"
 
 // Represents information about module on the client
 class ModuleData final {
  public:
   explicit ModuleData(orbit_grpc_protos::ModuleInfo info)
       : module_info_(std::move(info)), is_loaded_(false) {}
 
-  void SetModuleInfo(const orbit_grpc_protos::ModuleInfo& info) { module_info_ = info; }
+  [[nodiscard]] const std::string& name() const { return module_info_.name(); }
+  [[nodiscard]] const std::string& file_path() const { return module_info_.file_path(); }
+  [[nodiscard]] uint64_t file_size() const { return module_info_.file_size(); }
+  [[nodiscard]] const std::string& build_id() const { return module_info_.build_id(); }
+  [[nodiscard]] uint64_t load_bias() const { return module_info_.load_bias(); }
+  [[nodiscard]] bool is_loaded() const { return is_loaded_; }
 
-  const std::string& name() const { return module_info_.name(); }
-  const std::string& file_path() const { return module_info_.file_path(); }
-  uint64_t file_size() const { return module_info_.file_size(); }
-  uint64_t address_start() const { return module_info_.address_start(); }
-  uint64_t load_bias() const { return module_info_.load_bias(); }
-
-  std::string address_range() const {
-    return absl::StrFormat("[%016" PRIx64 " - %016" PRIx64 "]", module_info_.address_start(),
-                           module_info_.address_end());
+  // relative_address here is the absolute address minus the address this module was loaded at by
+  // the process (module start address)
+  [[nodiscard]] orbit_client_protos::FunctionInfo* FindFunctionByRelativeAddress(
```

Comment:
Yes, it would be a little cleaner right now, but I specifically designed it here like this for a certain reason. 
I want to decouple Module and Process in a way that Module is independent of the address that its loaded at (its base address / start address). I feel like this makes the structure and interfaces between these classes cleaner. A Process loads Modules in memory regions and Modules contain Functions. The Module and Functions are independent from the Process in the sense that a different Process could load the same Module and nothing would change. Heck, if we loaded the Symbols before, we can reuse them. 

This is at least what I want to work towards to. Currently FunctionInfo still contains the field `module_base_address` which makes this plan not possible for now. I added a couple of TODO comments about this. 

One benefit for example would be in that this structure enables us to support processes dynamically loading and unloading modules (even while we capture).

Reply:
Convinced me. Thanks for the clarification.